### PR TITLE
Update dev deps and fix code smells for V3

### PR DESCRIPTION
**Pull request by t1m0thyj**
## Proposed changes

Resolves security alerts and code smells found by GitHub now that V3 code is in the main branch.

## Release Notes

Milestone: 3.0.0

Changelog: N/A

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (non-breaking change which adds or improves functionality)
- [ ] Breaking change (a change that would cause existing functionality to not work as expected)
- [ ] Documentation (Markdown, README updates)
- [ ] Other (please specify above in "Proposed changes" section)

## Checklist

**General**

- [ ] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/zowe-explorer-vscode/wiki/Best-Practices:-Contributor-Guidance) wiki
- [ ] All PR dependencies have been merged and published (if applicable)
- [ ] A GIF or screenshot is included in the PR for visual changes
- [ ] The pre-publish command has been executed:
  - **v2 and below:** `yarn workspace vscode-extension-for-zowe vscode:prepublish`
  - **v3:** `pnpm --filter vscode-extension-for-zowe vscode:prepublish`

**Code coverage**

- [ ] There is coverage for the code that I have added
- [ ] I have added new test cases and they are passing
- [ ] I have manually tested the changes

**Deployment**

- [ ] I have added developer documentation (if applicable)
- [ ] Documentation should be added to Zowe Docs
  - If you're an outside contributor, please post in the [#zowe-doc Slack channel](https://openmainframeproject.slack.com/archives/CC961JYMQ) to coordinate documentation.
  - Otherwise, please check with the rest of the squad about any needed documentation before merging.
- [ ] These changes may need ported to the appropriate branches (list here):

### CODE REVIEW

**Review by JillieBeanSim — APPROVED** [review]
LGTM! thanks @t1m0thyj

File: packages/zowe-explorer-api/__mocks__/mockUtils.ts [hunk 1]
```diff
@@ -61,9 +61,7 @@ export class MockedProperty {
             return;
         }
 
-        const isValFn = typeof value === "function";
-
-        if (isValFn || (typeof descriptor?.value === "function" && value == null)) {
+        if (typeof value === "function") {
```
Comment (traeok) [inline packages/zowe-explorer-api/__mocks__/mockUtils.ts:64]: Thanks for cleaning this logic up 😋

**Review by traeok — APPROVED** [review]
Changes LGTM, thanks Timothy for the fixes & updates 😋

**Review by traeok — DISMISSED** [review]

**Review by JillieBeanSim — APPROVED** [review]

File: packages/zowe-explorer/scripts/generatePoeditorJson.js
Comment (traeok) [inline packages/zowe-explorer/scripts/generatePoeditorJson.js:1]: Question for my own clarification: could we remove some of this logic by defining the file suffix before looking over the entries in the JSON? E.g.:

```javascript
const fileSuffix = langId ? `.${langId}.json` : ".json";
```

Then the same file suffix could be appended to the `package.nls`, `poeditor` and `bundle.l10n` files to retrieve the contents for the given language ID. 

I didn't notice much of a difference between the two branches when doing the `langId` checks, but if we have to keep the logic separate then feel free to disregard this.

**Review by traeok — APPROVED** [review]
Had a quick question but nothing that should hold this PR back. Thanks Timothy for the updates & code smell fixes

**Pull request merged**